### PR TITLE
Replaced Element Tap Quiescence Check for iOS 13

### PR DESCRIPTION
**Pull request by Dan-Maor**
We've encountered a case in which using XCTest's native `[XCUIElement tap]` delays the command execution by 60 seconds compared to the custom implementation, which is not being used for iOS 13. After the 60 seconds delay a message stating "App animations complete notification not received, will attempt to continue" message is displayed in the device's log and the interaction succeeds.

Turns out that the function uses XCTest's quiescence validation internally (the same one being used for launching applications), and that it appears to not play nice with animations after interacting with elements.

From what I could gather, XCTest's animations check uses the `XCAXClient` method `notifyWhenNoAnimationsAreActiveForApplication`internally, which registers for a user testing notification titled "AnimationsNonActive", and then injects an accessibility event (2043) to the accessibility element of the application. This in turn would trigger the required user testing notification to be posted from the tested application.

It appears that for some reason attempting to check for animations after trying to find elements during an animation causes the application to not send the user testing notification required to finish the quiescence check, even if the check was initiated after the application has truly idled.

This PR disables the animation check before using `[XCUIElement tap]` and validates the readiness of the object to be tapped using `fb_waitUntilFrameIsStable`, like with the custom implementation.

It should be noted that when using Xcode 11, in some instances this approach is slightly slower than XCTest's implementation. I did not notice a speed discrepancy on Xcode 10.

### CODE REVIEW

**Comment by KazuCocoa** [comment]
ah...
it is out of my curiosity.
Does this slowness happen when `reduceMotion` on, too?
(it is available in caps and settings API for only on a simulator. On a real device, we must enable the preference in settings > accessibility by hand)

**Comment by Dan-Maor** [comment]
The issue is still happening when reduceMotion is enabled.

File: WebDriverAgentLib/Utilities/XCUIApplicationProcessQuiescence.h [hunk 1]
```diff
@@ -21,6 +21,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 + (void)setQuiescenceCheck:(BOOL)value;
 
+/**
+ Set the usage of animation check in quiescence validation (defaults to YES).
+ */
++ (void)setEnableAnimationCheck:(BOOL)value;
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Utilities/XCUIApplicationProcessQuiescence.h]: What about like below naming?

- `setAnimationCheckEnabled: (BOOL)isEnabled`
- `setEnableAnimationCheck: (BOOL)enable`

**Review by KazuCocoa — COMMENTED** [review]
Thanks.
This change looked good to me

**Comment by mykola-mokhnach** [comment]
wow, thanks for the interesting investigation Mr @Dan-Maor 
I saw similar reports at https://github.com/appium/appium/issues/13503#issuecomment-580206360 and I'm afraid this problem is not only relevant for tap, but also for other similar XCUIElement primitives.

File: WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m [hunk 2]
```diff
@@ -39,14 +39,16 @@ - (BOOL)fb_waitUntilFrameIsStable
...
     timeout:10.]
    spinUntilTrue:^BOOL{
-    const BOOL isSameFrame = FBRectFuzzyEqualToRect(self.frame, frame, FBDefaultFrameFuzzyThreshold);
-    frame = self.frame;
+    CGRect newFrame = self.frame;
+    const BOOL isSameFrame = FBRectFuzzyEqualToRect(newFrame, frame, FBDefaultFrameFuzzyThreshold);
+    frame = newFrame;
     return isSameFrame;
   }];
+  return result;
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m]: is this change necessary?

File: WebDriverAgentLib/Categories/XCUIElement+FBTap.m [hunk 3]
```diff
@@ -21,7 +22,10 @@ - (BOOL)fb_tapWithError:(NSError **)error
   if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"13.0")) {
     // Tap coordinates calculation issues have been fixed
     // for different device orientations since Xcode 11
+    [self fb_waitUntilFrameIsStable];
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Categories/XCUIElement+FBTap.m:28]: Please add an explanation about why this workaround is necessary (a link to this PR could be enough)

**Review by KazuCocoa — APPROVED** [review]
lgtm with Mykola's comment

File: WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m (hunk 2 above)
Comment (Dan-Maor) [reply WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m]: Calling `self.frame` retrieves a snapshot of the element every time.

From my testing saving the result of the frame instead of calling `self.frame` every time the property is needed is slightly faster (5% faster in my simple stress scenario); I've experimented with ways to make the possible performance impact lower, also tested using `fb_snapshotwithattributes` with only the frame as a parameter, but it didn't make much of a difference from just calling `self.frame`.

Saving the result into a variable was just for printing execution times though, I'll remove it.

**Review by mykola-mokhnach — APPROVED** [review]
